ethereum/p2p/src/main/java/org/hyperledger/besu/ethereum/p2p/rlpx/RlpxAgent.java

```diff
@@ -463,21 +471,29 @@ private void enforceConnectionLimits() {
   private Stream<RlpxConnection> getActivePrioritizedConnections() {
     return connectionsById.values().stream()
         .filter(RlpxConnection::isActive)
-        .sorted(this::prioritizeConnections);
+        .sorted(this::comparePeerPriorities);
   }
 
-  private int prioritizeConnections(final RlpxConnection a, final RlpxConnection b) {
+  private int comparePeerPriorities(final RlpxConnection a, final RlpxConnection b) {
     final boolean aIgnoresPeerLimits = peerPrivileges.canExceedConnectionLimits(a.getPeer());
     final boolean bIgnoresPeerLimits = peerPrivileges.canExceedConnectionLimits(b.getPeer());
     if (aIgnoresPeerLimits && !bIgnoresPeerLimits) {
       return -1;
     } else if (bIgnoresPeerLimits && !aIgnoresPeerLimits) {
       return 1;
     } else {
-      return Math.toIntExact(a.getInitiatedAt() - b.getInitiatedAt());
+      return randomPeerPriority ? compareRandomly(a, b) : compareConnectionInitiationTimes(a, b);
     }
   }
 
+  private int compareConnectionInitiationTimes(final RlpxConnection a, final RlpxConnection b) {
+    return Math.toIntExact(a.getInitiatedAt() - b.getInitiatedAt());
+  }
+
+  private int compareRandomly(final RlpxConnection a, final RlpxConnection b) {
```

Comment:
`Randomly` feels like the wrong name.  While the mask is random once it's set it doesn't change. `compareByMaskedNodeId`?

Reply:
Yeah, I felt similarly when writing it. I like your suggestion.